### PR TITLE
moving instance analytics logic to enterprise

### PR DESCRIPTION
**Pull request by npfitz**
Closes https://github.com/metabase/metabase/issues/31212

### Description
Moves logic to determine collection Icon to enterprise plugins

### How to verify
Verify that collections in the nav bar show the correct icons, and that the collection header also shows the beaker icon

### Demo
image

### Checklist
- [ ] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: enterprise/frontend/src/metabase-enterprise/collections/utils.ts
Comment (iethree) [inline enterprise/frontend/src/metabase-enterprise/collections/utils.ts:1]: we should probably move the `getCollectionTooltip` function from `entities/collections/utils` into this file as well.

**Review by iethree — CHANGES_REQUESTED** [review]
Approach looks good, I think we just need to move one other function as well. 👍

**Review by iethree — APPROVED** [review]
Nice. thanks for doing this rather tedious and boring bit of moving stuff around.

I think there's a couple unit tests that just need

```js
beforeEach(() => {
    setupEnterpriseTest();
});
```

**Review by oisincoveney — APPROVED** [review]
Just here for the `DashboardHeader` unit test changes, looks good 😄

**Pull request merged**